src/queryIterator.ts

```diff
@@ -217,4 +225,21 @@ export class QueryIterator<T> {
       error.substatus === SubStatusCodes.CrossPartitionQueryNotServable
     );
   }
+
+  private initPromise: Promise<void>;
```

Comment:
Discussed with you over teams. Lmk what you wanna do.

Reply:
Added new option to make this behavior change opt-in